### DISCUSSION TITLE
web API extension

### DISCUSSION THREAD

**Discussion by Schlonsy** [tbnobody/OpenDTU#832] [Ideas]
Hallo Zusammen,

vielleicht wäre es hilfreich wenn man von außen auf möglichst einfachem Weg Änderungen an der API erkennen könnte.
Ich schlage daher vor die API um einer Versions-Information zu erweitern (z.B. /api/version). Ein Integer Wert fortlaufend hochgezählt bei Änderungen der API. So könnten Anwender der API leicht erkennen, ob sie kompatibel mit der API sind und ggf. Änderungen auf ihrer Seite davon ableiten, ohne tief in die API eindringen zu müssen.

Wie seht ihr das?

Viele Grüße,
    Karsten

**Comment by h3tz** [comment]
Das ist ja gerade Sinn und Zweck einer API das sich NIX ändert an bestehendem. Neues kann dazu kommen.

**Reply by Schlonsy** [reply]
Hallo Andreas,
bisher habe ich für eine API stets einen anderen Sinn und Zweck gesehen und kenne mehr APIs die im Laufe der Zeit inkompatible Änderungen erfahren mussten, als welche die nur Neues hinzugefügt haben. Aber ich möchte damit keine Diskussion über Sinn und Zweck einer API entfachen, sondern mir kam es lediglich sinnvoll vor den Stand einer API auf einfachen Weg in Erfahrung bringen zu können.
War nur eine Idee, die ich beitragen wollte, hätte ja eine gute sein können ;-).
Viele Grüße,
Karsten

**Reply by h3tz** [reply]
Hoi. Wenn eine Versionsnummer dein Problem löst dann bleib bei dem Vorschlag. Es gibt allerdings auch die Möglichkeit test request auszusenden und bei dir intern etwas ähnliches zu tun.
